src/app/service/timer-service.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, of, reduce, scan, tap } from 'rxjs';
+
+export type Timer = {
+  label: string | null;
+  hours: number | null;
+  minutes: number | null;
+  seconds: number | null;
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TimerService {
+  private timers$ = new BehaviorSubject<Timer[]>([]);
+
+  constructor() {}
+
+  getTimers(): Observable<Timer[]> {
+    return this.timers$.asObservable();
+  }
+
+  addTimer(timer: Timer) {
+    this.timers$.next([...this.timers$.value, timer]);
```

Comment:
Can we do this.timers$.next(timer); and use scan operator to reduce it?

Reply:
Throws error: Argument of type 'Timer' is not assignable to parameter of type 'Timer[]'

Reply:
BEcause it should be type of Timer and reducer should return Timer[]